File: proto-csharp/ProtoMessage/ProtoMessage/ApeMessage/Reflect/DoubleFieldDescriptor.cs
﻿using System;
using System.Reflection;
using System.Collections.Generic;
using Google.ProtocolBuffers;
using com.spaceape.protobuf;

namespace com.spaceape.protobuf.reflect
{
  public class DoubleFieldDescriptor: FieldDescriptor
  {
    public DoubleFieldDescriptor(int fieldNumber, String fieldName, Type clazz) : base(fieldNumber, fieldName, clazz)
    {
    }

    override public Object readField(CodedInputStream input, CodeGenContext context)
    {
      return input.ReadDouble();
    }

    override public void writeField(CodedOutputStream output, CodeGenContext context, Object obj)
    {
      output.WriteDouble(fieldNumber, (double)obj);
    }
  }
}

File: proto-java/src/main/java/com/spaceape/protobuf/reflect/DoubleFieldDescriptor.java
package com.spaceape.protobuf.reflect;

import com.google.protobuf.CodedInputStream;
import com.google.protobuf.CodedOutputStream;
import com.spaceape.protobuf.CodeGenContext;

import java.io.IOException;

public class DoubleFieldDescriptor extends FieldDescriptor{
    public DoubleFieldDescriptor(int fieldNumber, String fieldName, Class clazz){
        super(fieldNumber, fieldName, clazz, double.class);
    }
    public Double readField(CodedInputStream input, CodeGenContext context) throws IOException {
        return input.readDouble();
    }
    public void writeField(CodedOutputStream output, CodeGenContext context, Object object) throws IOException{
        output.writeDouble(fieldNumber, (double)object);
    }
}
